File: persistence/src/main/java/za/co/entelect/bootcamp/flash/persistence/Repository.java
package za.co.entelect.bootcamp.flash.persistence;

import za.co.entelect.bootcamp.flash.domain.Entity;

import java.util.List;

public interface Repository<TKey, TEntity extends Entity<TKey>> {

    void create(TEntity entity);

    TEntity getByID(TKey entityKey);

    List<TEntity> getAll();

    void update(TEntity entity) throws Exception;

    void delete(TEntity entity) throws Exception;
}

File: C#/ComicStock/ComicStock.WebAPI/Models/Repository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ComicStock.WebAPI.Models
{
    public abstract class Repository<TKey, TEntity> : IRepository<TKey, TEntity>
    {
        public void create(TEntity entity)
        {
            throw new NotImplementedException();
        }

        public void delete(TEntity entity)
        {
            throw new NotImplementedException();
        }

        public TEntity getById(TKey id)
        {
            throw new NotImplementedException();
        }

        public List<TEntity> getList()
        {
            throw new NotImplementedException();
        }

        public TEntity update(TEntity entity)
        {
            throw new NotImplementedException();
        }
    }
}